Helper/Data.php

```diff
@@ -142,60 +152,60 @@ public function getKlaviyoLists($api_key=null){
         return $result;
     }
 
-    public function subscribeEmailToKlaviyoList($email, $first_name=null, $last_name=null, $source=null) {
-        $list_id = $this->getNewsletter();
-        $opt_in = $this->getOptInSetting();
-        $api_key = $this->getPrivateApiKey();
+    /**
+     * @param string $email
+     * @param string $firstName
+     * @param string $lastName
+     * @param string $source
+     * @return bool|string
+     */
+    public function subscribeEmailToKlaviyoList($email, $firstName = null, $lastName = null, $source = null)
+    {
+        $listId = $this->getNewsletter();
+        $optInSetting = $this->getOptInSetting();
+        $apiKey = $this->getPrivateApiKey();
 
         $properties = [];
         $properties['email'] = $email;
-        if ($first_name) $properties['$first_name'] = $first_name;
-        if ($last_name) $properties['$last_name'] = $last_name;
+        if ($firstName) $properties['$first_name'] = $firstName;
+        if ($lastName) $properties['$last_name'] = $lastName;
         if ($source) $properties['$source'] = $source;
 
-        $properties_val = count($properties) ? json_encode(array('profiles' => $properties)) : '{}';
+        $propertiesVal = ['profiles' => $properties];
 
-        $url = "https://a.klaviyo.com/api/v2/list/" . $list_id . $opt_in . "?api_key=" . $api_key;
+        $path = self::LIST_V2_API . $listId . $optInSetting . "?api_key=" . $apiKey;
 
-        $curl = curl_init();
-        curl_setopt_array($curl, array(
-            CURLOPT_URL => $url,
-            CURLOPT_RETURNTRANSFER => true,
-            CURLOPT_CUSTOMREQUEST => "POST",
-            CURLOPT_POSTFIELDS => $properties_val,
-            CURLOPT_USERAGENT => self::USER_AGENT,
-            CURLOPT_HTTPHEADER => array(
-              "Content-Type: application/json",
-              'Content-Length: ' . strlen($properties_val)
-            ),
-        ));
-        // Submit the POST request
-        $response = curl_exec($curl);
-        $err = curl_error($curl);
-        // Close cURL session handle
-        curl_close($curl);
+        try {
+            $response = $this->sendApiRequest($path, $propertiesVal, 'POST');
+        } catch (\Exception $e) {
+            $this->logger->warning(sprintf('Unable to subscribe %s to list %s: %s', $email, $listId, $e));
+        }
 
         return $response;
     }
 
-    public function unsubscribeEmailFromKlaviyoList($email) 
+    /**
+     * @param string $email
+     * @return bool|string
+     */
+    public function unsubscribeEmailFromKlaviyoList($email)
     {
-        $list_id = $this->getNewsletter();
-        $api_key = $this->getPrivateApiKey();
+        $listId = $this->getNewsletter();
+        $apiKey = $this->getPrivateApiKey();
 
+        $path = self::LIST_V2_API . $listId . self::API_SUBSCRIBE;
         $fields = [
-            'api_key=' . $api_key,
-            'email=' . urlencode($email),
+            'api_key' => (string)$apiKey,
```

Comment:
sorry one more thing (but looks good 👍), i know our api accepts two different types of auth either utm param of api_key or api_key in the json, could we move this into the sendApiRequest and add it as a param?

Reply:
Why we need it? I did it as showed in the docs example:
https://www.klaviyo.com/docs/api/v2/lists#delete-subscribe
![image](https://user-images.githubusercontent.com/1873745/72985265-46f20b80-3dee-11ea-958f-87cba84c2573.png)

Is there any issues or benefits of doing that?

Reply:
Or you meant that we can add api key param inside `sendApiRequest`,
like 
```php
private function sendApiRequest(string $path, array $params, string $method = 'POST')
    {
        $params['api_key'] = $this->getPrivateApiKey(); // <!-- see to this line
        $url = self::KLAVIYO_HOST . $path;

        $curl = curl_init();
        ...

Reply:
yes please to this one 👍

Reply:
Done